### PR TITLE
new high contrast styles for slider steps

### LINKED ISSUE #23149: v9 Slider: Windows contrast mode ticks are low contrast
Slider uses `Highlight` for the track background in HCM, but the tick marks are regular text/border color, so the result is a fairly low-contrast set of ticks (depending on the theme of course, but `CanvasText` isn't really expected to be visible against `Highlight`).

screenshot of a slider with white tick marks against a light blue track

We should probably update the tick color to be `HighlightText` in forced-colors.

### PR DESCRIPTION
**Pull request by micahgodbolt**
fixes #23149

Add specific high contrast styles for slider steps to ensure that they are visible.

before
image

after

image

### CODE REVIEW

File: packages/react-components/react-slider/src/components/Slider/useSliderStyles.ts [hunk 1]
```diff
@@ -146,7 +146,17 @@ const useRailStyles = makeStyles({
...
+        backgroundImage: `repeating-linear-gradient(
+          var(${railDirectionVar}),
+          #0000 0%,
+          #0000 calc(var(${railStepsPercentVar}) - 1px),
+          HighlightText calc(var(${railStepsPercentVar}) - 1px),
+          HighlightText var(${railStepsPercentVar})
+        )`,
+      },
     },
+    // force steps to use HighlightText for high contrast mode
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-slider/src/components/Slider/useSliderStyles.ts:159]: Nit: Did the comment get put after the code?

File: packages/react-components/react-slider/src/components/Slider/useSliderStyles.ts [hunk 2]
```diff
@@ -146,7 +146,17 @@ const useRailStyles = makeStyles({
         ${tokens.colorNeutralBackground1} calc(var(${railStepsPercentVar}) - 1px),
         ${tokens.colorNeutralBackground1} var(${railStepsPercentVar})
       )`,
+      '@media (forced-colors: active)': {
+        backgroundImage: `repeating-linear-gradient(
+          var(${railDirectionVar}),
+          #0000 0%,
```
Comment (GeoffCoxMSFT) [inline packages/react-components/react-slider/src/components/Slider/useSliderStyles.ts:152]: Four zeros isn't something I've seen before, did you mean #000

**Review by GeoffCoxMSFT — APPROVED** [review]

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                          |    Baseline (minified/GZIP) |                          PR |                                                                                                                                                                                                  Change |
| :----------------------------------------------------------------------------------------- | --------------------------: | --------------------------: | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-slider</samp> <br /> <abbr title='bundle-size/Slider.fixture.js'>Slider</abbr> | `25.557 kB`<br />`8.266 kB` | `25.963 kB`<br />`8.314 kB` | `406 B` <br />`48 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `181.156 kB`<br />`50.608 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.078 kB`<br />`11.098 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/e0417c44ab690bda8923df041a880b6085f48e2d'>e0417c44ab690bda8923df041a880b6085f48e2d</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/Avatar-mount_base.html">1275</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/Avatar-mount.html">1282</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/Button-mount_base.html">811</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/Button-mount.html">779</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProvider-mount_base.html">2338</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProvider-mount.html">2335</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProviderWithTheme-mount_base.html">430</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProviderWithTheme-mount.html">414</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">453</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">384</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">464</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">451</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/MakeStyles-mount_base.html">2068</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23253/perf-test-react-components/MakeStyles-mount.html">2074</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Pull request merged**